### ISSUE TITLE
Clicking on a product only loads browser after 2-3 secondes.

### ISSUE THREAD

**Issue by RomainLaf** [duracelltomi/gtm4wp#337]
Hello,

I'm facing an issue : When a user clicks on a product (home page or shop page) to go to product page detail (single post page) It seems like GTM4WP is blocking the page loading. 

Here's the step I go through (Screenrecord : https://share.cleanshot.com/ZrV9TGZH)

When I click on a product at **0:15** the browser doesn't even start loading anything. 
It only starts loading at **0:17** (you can see that the loading icon in the URL bar starts showing at 0:17 (~2/3seconds after the click)) https://share.cleanshot.com/YY7LQzvG

In the same screen, you can see at **0:17** that the first element to load in the Network tab of Chrome Dev Tool is something about GTM : https://share.cleanshot.com/YY7LQzvG

It's very strange because everything else works fine. The website is super fast and loading any pages works well.

I saw an another (more or less similar) issue report that someone posted, talking about Safari and product page loads. 
[Problem on safari with wp-rocket and gtm4wp](https://wordpress.org/support/topic/problem-on-safari-with-wp-rocket-and-gtm4wp/page/2/ ) and you seems to fix this issue with the commit 96a936a
But on my side, the issue happens on Chrome, Safari, Firefox... 

The only different state is that on private browsers the product page start loading super fast, right after user clicks. Here's a screen record example on iOS Safari (but Safari is not the issue because as I said, it happens on Desktop and mobile and with Chrome or safari). So I don't know why this issue doesn't appear on private browsing. Does the script loads differently ? With less or different data or trigger ?

I talked with Kinsta support (where my website is hosted). And they pointed to me that it's GTM4WP that causes this issue. So they asked me to test in a Staging Environment to deactivate GTM4WP. And voila! The issue is gone and the product page start loading right after user clicks. (The browser doesn't wait 2-3 seconds to start loading).

I'm not an advanced dev, so I don't know how to fix this issue and where to have a look. I just now that without GTM4WP the issue is gone. 

Please can you help ?

**Comment by RomainLaf** [comment]
I don't know if I'm clear on the issue and if you can see what I'm trying to point out... 
Here's a new screen record. One with safari iOS (first step is with private browsing and second step with normal browsing tab) : https://share.cleanshot.com/bBMYDM3X

And two other ones with Chrome. Both steps with Chrome's normal browsing tab.
The first one is with GTM4WP plugin activated : https://share.cleanshot.com/lDMTyhjk
The second one is without GTM4WP (plugin deactivated) : https://share.cleanshot.com/P3P1Wz4y

**Comment by JoshBour** [comment]
I think I have the same issue, and I can give some specific information @duracelltomi 
When the plugin is loaded with "Google Consent Mode" **disabled**, when you click on a product it fires the following events (as seen in the GTM Debugger):
1) FIRST it fires the "Link Click" event
2) THEN it fires the "select_item" event

If you **enable** the "Google Consent Mode", without checking any other boxes, when you click on the same product it fires them in reverse:
1) FIRST it fires the "select_item" event
2) THEN it fires the "Link Click" event.

My guess is that because the "select_item" is fired first, until the request is completed the browser doesn't respond to the click.
I don't know why it triggers in the wrong order when the Google Consent Mode is enabled, but I guess it's an easy fix.

**Comment by RomainLaf** [comment]
I'm not sure about what Google Consent Mode is and how to use it, so on my side those setting's tab checkboxes are all unchecked. 
So this doesn't seem to be the source of my issue.

**Comment by RomainLaf** [comment]
Thanks for the time taken and all deep technical background. Explaining the 2000ms timeout helped me to figure out that it was a GTM issue and not GTM4WP.
I tried as you mention using GTM ID GTM-XXXXXX and the issue was gone, so I dived into GTM Assistant and find out that a tag was blocking and never succeeded ("still running"). It was a "Google Ads Remarketing" tag : https://share.cleanshot.com/QJn593fp

Disabling it solved my issue. I don't know why and I don't know what was wrong with this tag. But now my website is no longer blocking item select with 2000ms timeout. 

Thanks again for your help and time taken !
